packages/legend-application-studio/src/components/editor/edit-panel/mapping-editor/MappingTestEditor.tsx

```diff
@@ -103,8 +106,14 @@ const MappingTestQueryEditor = observer(
     const applicationStore = useApplicationStore();
 
     // actions
-    const editWithQueryBuilder = applicationStore.guardUnhandledError(
-      async () => {
+    // double-click read-only query tab to open query builder either in form mode or text mode by
+    // controlling the value of openQueryBuilderTextEditor
+    // support an option to always edit query in text mode by setting alwaysOpenQueryBuilderTextEditor to true
+    const editWithQueryBuilder = (
+      openQueryBuilderTextEditor: boolean,
```

Comment:
these flag names are very confusing. please rename for they actually give a description of why they are needed.